sale_order_line_input/models/sale_order.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    pricelist_id = fields.Many2one(
+        related="order_id.pricelist_id",
+        readonly=True,
+    )
+    with_company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Forced company",
+        compute="_compute_with_company_id",
+        readonly=False,
+        help="Technical field to force company or get it "
+        "from env user if order don't exist.",
+    )
+
+    @api.depends("order_id")
+    def _compute_with_company_id(self):
+        """Related company is not computed already when we click create new line"""
+        for line in self:
+            line.with_company_id = (
+                line.order_id.company_id
+                # Is not necessary use browse here
+                or self.env.context.get("with_company")
+                or self.env.company
+            )
+
+    @api.onchange("with_company_id")
+    def _onchange_with_company_id(self):
+        """Assign company_id because is used in domains as partner,
+        product, taxes..."""
+        for line in self:
+            line.company_id = line.with_company_id
+
+    @api.onchange("order_partner_id")
+    def _onchange_order_partner_id(self):
+        """Create order to correct compute of taxes"""
+        if not self.order_partner_id or self.order_id:
+            return
+        SaleOrder = self.with_context(with_company=self.with_company_id.id).env[
```

Comment:
https://www.odoo.com/documentation/15.0/es/developer/howtos/company.html

```suggestion
        SaleOrder = self.with_company(self.force_company_id).env[
```